mintapi/api.py

```diff
@@ -349,42 +352,60 @@ def _sign_in(email, password, driver, mfa_method=None, mfa_token=None,
-                        mfa_method_option = driver.find_element_by_id(
-                            'ius-mfa-option-{}'.format(mfa_method))
-                        mfa_method_option.click()
-                    except NoSuchElementException:
-                        logger.info(
-                            'sms button is not there. Trying without selection')
+                    driver.find_element_by_id('ius-mfa-options-form')
+                    # Not sure if this method is still works some of the time. It could be that this has been deprecated/changed by mint.com
+                    mfa_method_option = driver.find_element_by_id(
+                        'ius-mfa-option-{}'.format(mfa_method))
+                    mfa_method_option.click()
                     mfa_method_submit = driver.find_element_by_id(
                         "ius-mfa-options-submit-btn")
                     mfa_method_submit.click()
+                except NoSuchElementException:
+                    pass # no option to select mfa option
 
+                try:
+                    mfa_code_input = driver.find_element_by_id("ius-mfa-confirm-code")
+                    mfa_code_input.clear()
                     if mfa_method == 'email' and imap_account:
                         mfa_code = get_email_code(imap_account, imap_password, imap_server, imap_folder=imap_folder)
                     else:
                         mfa_code = (mfa_input_callback or input)("Please enter your 6-digit MFA code: ")
-                    mfa_code_input = driver.find_element_by_id("ius-mfa-confirm-code")
                     mfa_code_input.send_keys(mfa_code)
 
                     mfa_code_submit = driver.find_element_by_id("ius-mfa-otp-submit-btn")
                     mfa_code_submit.click()
-                except Exception:  # if anything goes wrong for any reason, give up on MFA
-                    mfa_method = None
-                    logger.warning("Giving up on handling MFA. Please "
-                                   "complete the MFA process manually in the "
-                                   "browser.")
+                except (NoSuchElementException, ElementNotInteractableException):
+                    pass # we're not on mfa input screen
+
         except NoSuchElementException:
-            pass
+            pass # not on mfa screen
+
+        # account selection screen -- if there are multiple accounts, select one
+        try:
+            select_account = driver.find_element_by_id("ius-mfa-select-account-section")
+            if intuit_account is not None:
+                account_input = select_account.find_element_by_xpath(
+                    "//label/span[text()='{}']/../preceding-sibling::input".format(intuit_account))
+                account_input.click()
+            driver.find_element_by_id("ius-sign-in-mfa-select-account-continue-btn").click()
+            time.sleep(2)  # hack... what can we consistently waitfor?
```

Comment:
Is it possible to find an element to wait for here? This is going to be a bit brittle depending on internet connections, latency, etc.

Reply:
agree... changed it from a click() to a submit().  the latter waits for the button's form to complete.  verified that still works with my multiple accounts.